smart_open/gcs.py

```diff
@@ -0,0 +1,524 @@
+# -*- coding: utf-8 -*-
+"""Implements file-like objects for reading and writing to/from GCS."""
+
+import io
+import logging
+import sys
+
+from google.cloud import storage, exceptions
+import google.auth.transport.requests as google_requests
+import six
+
+import smart_open.bytebuffer
+
+logger = logging.getLogger(__name__)
+
+READ_BINARY = 'rb'
```

Comment:
These can probably be internal.

Please review all module-level variables and functions. If they aren't needed from outside this module, then make them internal. If they are needed outside, add descriptive docstrings.